twisted/mail/imap4.py

```diff
@@ -460,7 +460,7 @@ def newMessages(exists, recent):
 # Some definitions (SP, CTL, DQUOTE) are also from the ABNF RFC -
 # <https://tools.ietf.org/html/rfc2234>.
 _SP = ' '
-_CTL = ''.join(chr(ch) for ch in range(0x21) + range(0x80, 0x100))
+_CTL = ''.join(chr(ch) for ch in list(range(0x21)) + list(range(0x80, 0x100)))
```

Comment:
This works.  Another way to fix this is to do something like:

```
from itertools import chain

_CTL = ''.join(chr(ch) for ch in chain(range(0x21),  range(0x80, 0x100)))
```

That allows you to combine the ranges into one iterable without allocating multiple lists.